v2/pkg/engine/resolve/loader.go

```diff
@@ -936,11 +936,37 @@ func (l *Loader) optionallyOmitErrorFields(values []*astjson.Value) {
 
 // optionallyOmitErrorLocations removes the "locations" object from all values.
 func (l *Loader) optionallyOmitErrorLocations(values []*astjson.Value) {
-	if !l.omitSubgraphErrorLocations {
-		return
-	}
 	for _, value := range values {
-		if value.Exists("locations") {
+		if !value.Exists("locations") {
+			continue
+		}
+
+		// If the flag is set, delete all locations
+		if l.omitSubgraphErrorLocations {
+			value.Del("locations")
+			continue
+		}
+
+		// Parse locations into typed structure for validation
+		locationsJSON := value.Get("locations").MarshalTo(nil)
+		var parsedLocations []ValidationLocation
+		if err := json.Unmarshal(locationsJSON, &parsedLocations); err != nil {
```

Comment:
Tbh not sure if its a good idea to parse again on the hot path in case locations are there

Reply:
can you not just access the values and validate them? if invalid, omit